src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -688,7 +688,10 @@ namespace winrt::TerminalApp::implementation
     {
         if (auto presenter{ _dialogPresenter.get() })
         {
-            presenter.ShowDialog(FindName(L"AboutDialog").try_as<WUX::Controls::ContentDialog>());
+            if (presenter.CanShowDialog())
+            {
+                presenter.ShowDialog(FindName(L"AboutDialog").try_as<WUX::Controls::ContentDialog>());
+            }
```

Comment:
Wouldn't it make the code more robust if we checked for `presenter.CanShowDialog()` inside `ShowDialog()`?

Reply:
That won't fix the bug because the call we want to avoid making if there's already a `ContentDialog` being shown is the `FindName` call.

Also, like the logical separation between `ShowDialog` and `CanShowDialog`, instead of having `ShowDialog` do double duty of checking whether a dialog is currently open before showing it. It keeps the presenter as a sort of 'dumb' component and puts the onus on the caller.

Reply:
> [...] because the call we want to avoid making if there's already a `ContentDialog` being shown is the `FindName` call.

The call to `FindName` causes the XAML content to move? Wat. 😅

I'd probably still write an abstraction here, like this:
```cpp
void TerminalPage::_showDialog(std::wstring_view name)
{
    if (presenter.CanShowDialog())
    {
        co_return co_await presenter.ShowDialog(FindName(name).try_as<WUX::Controls::ContentDialog>());
    }
}
```

But I'm also fine with the current solution.

Reply:
>The call to FindName causes the XAML content to move? Wat. 

_YEP_

>I'd probably still write an abstraction here, like this:

That's a good idea, added!